Storage/api_base.h

```diff
@@ -33,6 +33,7 @@ namespace rest {
 
 template <typename STORAGE>
 struct RESTfulGenericInput {
+  using STORAGE_TYPE = STORAGE;
```

Comment:
This is to alleviate the awkwardness of `ImmutableFields<std::remove_reference<decltype(in.storage)>::type>`